src/basic/memstream-util.c

```diff
@@ -0,0 +1,69 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "memstream-util.h"
+
+typedef struct MemStream {
+        FILE *f;
+        char *buf;
+        size_t sz;
+} MemStream;
+
+MemStream* memstream_free(MemStream *m) {
+        if (!m)
+                return NULL;
+
+        /* First, close file stream, as the buffer may be reallocated on close. */
+        safe_fclose(m->f);
+
+        /* Then, free buffer. */
+        free(m->buf);
+
+        return mfree(m);
+}
+
+FILE* memstream_open(MemStream **ret) {
```

Comment:
hmm, maybe call memstream_allocate()? for me "open" actually sounds more appropriate if we open something that exists already, but here we allocate something entirely new.

Reply:
Renamed to `memstream_init()`, as it does not allocate anything at least explicitly now.